### PR TITLE
Ignorable Nag Dialogs

### PR DESCRIPTION
**Pull request by Windchild292**
This PR swaps our nag dialogs to the new AbstractMHQNagDialog, which is backed onto MekHQ Options to allow persistence in ignoring individual nags

It also fixes Campaign::getDeploymentDeficit.

### CODE REVIEW

File: MekHQ/src/mekhq/gui/dialog/InsufficientAstechTimeNagDialog.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    //endregion Constructors
+
+    @Override
+    protected boolean checkNag(Campaign campaign) {
+        if (MekHQ.getMekHQOptions().getNagDialogIgnore(getKey())
+                || !campaign.getCampaignOptions().checkMaintenance()) {
+            return false;
+        }
+
+        final int need = campaign.getHangar().getUnitsStream()
```
Comment (NickAragua) [inline MekHQ/src/mekhq/gui/dialog/InsufficientAstechTimeNagDialog.java:44]: Let's put a comment explaining this syntax, as it's a little lengthy and takes some time to decode.

File: MekHQ/src/mekhq/gui/dialog/OutstandingScenariosNagDialog.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public OutstandingScenariosNagDialog(final JFrame frame, final Campaign campaign) {
+        super(frame, "OutstandingScenariosNagDialog", "OutstandingScenariosNagDialog.title",
+                "OutstandingScenariosNagDialog.text", campaign,
+                MekHqConstants.NAG_OUTSTANDING_SCENARIOS);
+    }
+    //endregion Constructors
+
+    @Override
+    protected boolean checkNag(final Campaign campaign) {
+        return !MekHQ.getMekHQOptions().getNagDialogIgnore(getKey())
```
Comment (NickAragua) [inline MekHQ/src/mekhq/gui/dialog/OutstandingScenariosNagDialog.java:39]: Let's put a comment explaining this.

**Review by NickAragua — APPROVED** [review]
Request for documentation on some of the more gnarly-looking stream statements. This syntax reminds me of stuff like using bit-shifting for division and other kinds of "old-school" tricks like that: it's pretty clever, but hard to parse if you're not "in the moment".

File: MekHQ/src/mekhq/gui/CampaignGUI.java [hunk 3]
```diff
@@ -2591,43 +2557,61 @@ private void refreshPartsAvailability() {
...
         }
+
         if (getCampaign().checkYearlyRetirements()) {
             showRetirementDefectionDialog();
-            ev.cancel();
+            evt.cancel();
+            return;
+        }
+
+        if (new UnmaintainedUnitsNagDialog(getFrame(), getCampaign()).showDialog().isCancelled()) {
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/gui/CampaignGUI.java:2582]: Random thought: a helper class with static methods could clean these up (I'd then throw them all in there as static classes):
```java
if (!Nags.checkForUnmaintainedUnits(getFrame(), getCampaign())) {
```

Comment (Windchild292) [reply MekHQ/src/mekhq/gui/CampaignGUI.java:2582]: I'm not the biggest fan of statics, and... it would be another class call for a single line, which is rarely worth it.
